File: kimssumin/Q27.py
def solution(progresses, speeds):
    answer = []
    remain_day = []
    for idx in range(len(progresses)):
        if (100 - progresses[idx]) % speeds[idx] == 0:
            day = (100 - progresses[idx]) // speeds[idx]
            remain_day.append(day)
        else:
            day = (100 - progresses[idx]) // speeds[idx] + 1
            remain_day.append(day)
    base_idx = 0
    for i in range(len(remain_day)):
        if remain_day[base_idx] < remain_day[i]: 
             # 둘의 차이 = 배포 개수
            answer.append(i - base_idx)
            base_idx  = i 

    answer.append(len(remain_day) - base_idx)

    return answer

File: BaeYoungSuk/Q27.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> list = new LinkedList<>();
        Queue<Integer> progressQueue = new LinkedList<>();
        Queue<Integer> speedQueue = new LinkedList<>();
        
        for(int i = 0; i < progresses.length; i++){
            progressQueue.offer(progresses[i]);
        }
        
        for(int i = 0; i < speeds.length; i++){
           speedQueue.offer(speeds[i]); 
        }
        
        int day = 1;
        while(!progressQueue.isEmpty()){
            int count = 0;
            
            if(progressQueue.peek() + (day * speedQueue.peek()) < 100){
                day++;
                continue;
            }
            
            while((!progressQueue.isEmpty()) && (!speedQueue.isEmpty())){
                if(progressQueue.peek() + (day * speedQueue.peek()) < 100) break;
                progressQueue.poll();
                speedQueue.poll();
                count++;
            }
            
            list.add(count);
        }
        
        int[] answer = list.stream()
            .mapToInt(i -> i)
            .toArray();   
          
        return answer;
    }
    
}
